src/factories/guide.ts

```diff
@@ -27,23 +30,30 @@ export async function guideFactory() {
   const label = inter('')
   element.addChild(label)
 
+  let scale = await waitForScale()
   let currentDate: Date | undefined
   let currentLabelFormatter: FormatDate
 
+  emitter.on('scaleUpdated', updated => scale = updated)
```

Comment:
This keeps scales up to date without having to make the functions that depend on them async.